archiver/archive.go

```diff
@@ -0,0 +1,112 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package archiver
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+// ArchivePackage method builds and streams an archive with package content.
+func ArchivePackage(w io.Writer, packagePath string) error {
+	gzipWriter := gzip.NewWriter(w)
+	tarWriter := tar.NewWriter(gzipWriter)
+	defer func() {
+		err := tarWriter.Close()
+		if err != nil {
+			log.Printf("Error occurred while closing tar writer: %v", err)
+		}
+
+		err = gzipWriter.Close()
+		if err != nil {
+			log.Printf("Error occurred while closing gzip writer: %v", err)
+		}
+	}()
+
+	err := filepath.Walk(packagePath, func(path string, info os.FileInfo, err error) error {
```

Comment:
Should we have some logic to skip hidden files that might be created by the system like `.DS_Store`?

Reply:
I wouldn't pollute the code with references to files that are ".gitignored" anyway. The docker image is created and pushed in the CI environment, I don't think we can be afraid of presence of such files there.

Reply:
This code is not only run inside docker but also locally.